llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -2980,6 +2994,19 @@ LegalizerHelper::widenScalar(MachineInstr &MI, unsigned TypeIdx, LLT WideTy) {
     Observer.changedInstr(MI);
     return Legalized;
   }
+  case TargetOpcode::G_VSCALE: {
+    MachineOperand &SrcMO = MI.getOperand(1);
+    LLVMContext &Ctx = MIRBuilder.getMF().getFunction().getContext();
+    const APInt &SrcVal = SrcMO.getCImm()->getValue();
+    // The CImm is always a signed value
+    const APInt &Val = SrcVal.sext(WideTy.getSizeInBits());
```

Comment:
Nit, this can be `APInt Val =`. The sext creates a new APInt, so you're taking a reference to a temporary object.